File: 15000/BOJ_15922 아우으 우아으이야!!.java
// https://www.acmicpc.net/problem/15922
import java.util.*;

public class Main {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        
        int a, b;
        int left = 0, right = 0;
        int sum = 0;

        for(int i = 0; i < n; ++i) {
            a = sc.nextInt();
            b = sc.nextInt();

            if(i == 0) {
                left = a;
                right = b;
            }
            else {
                if(a > right) {
                    sum += right - left;
                    left = a;
                    right = b;
                }
                else right = Math.max(b, right);
            }
        }

        System.out.println(sum + right - left);
    }
}

File: 15000/BOJ_15922 아우으 우아으이야!!.c
// https://www.acmicpc.net/problem/15922
#include <stdio.h>

int left = -1e9, right = -1e9;
int sum = 0;

int max(int x, int y) { return x > y ? x : y; }

int main()
{
    int n;
    scanf("%d", &n);

    for(int i = 0; i < n; ++i) {
        int x, y;
        scanf("%d %d", &x, &y);

        if(x > right) {
            sum += right - left;
            left = x;
            right = y;
        }
        else {
            right = max(y, right);
        }
    }

    printf("%d", sum + right - left);
}